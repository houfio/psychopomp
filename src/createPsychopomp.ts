import { createDakpan } from 'dakpan';
import { createElement, ReactNode } from 'react';

import { Psychopomp as Component } from './Psychopomp';

export const createPsychopomp = () => {
  const { Provider, withDakpan } = createDakpan({
    index: 0,
    steps: []
  })({
    nextIndex: () => ({ index, steps }) => ({
      index: Math.min(index + 1, steps.length - 1)
    }),
    previousIndex: () => ({ index }) => ({
      index: Math.max(0, index)
    })
  });

  const Psychopomp = ({ children }: { children: ReactNode }) => createElement(
    Provider,
    null,
    createElement(withDakpan(() => ({}))(Component)),
    children
  );

  return {
    Psychopomp,
    withPsychopomp: withDakpan
  };
};
