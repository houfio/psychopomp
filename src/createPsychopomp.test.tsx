import { mount } from 'enzyme';
import * as React from 'react';

import { createPsychopomp } from './createPsychopomp';

it('does stuff', () => {
  const { Psychopomp } = createPsychopomp();

  const wrapper = mount(
    <Psychopomp>
      <span>Test</span>
    </Psychopomp>
  );

  expect(wrapper).toMatchSnapshot();
});
