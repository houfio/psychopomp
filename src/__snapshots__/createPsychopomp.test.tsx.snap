// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does stuff 1`] = `
<Psychopomp>
  <Provider>
    <WithDakpan>
      <Consumer>
        <Component>
          <span>
            Psychopomp
          </span>
        </Component>
      </Consumer>
    </WithDakpan>
    <span>
      Test
    </span>
  </Provider>
</Psychopomp>
`;
